[AOTInductor] Allow using ProxyExecutor for ATen fallbacks

SherlockNoMad opened the pull request:
Summary: Use ProxyExecutor for aten._scaled_dot_product_efficient_attention in ABI-mode

Test Plan: OSS CI

Differential Revision: D51005807




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

chenyang78 reviewed (changes requested):
Fix the failing tests?

SherlockNoMad committed:
[AOTInductor] Allow using ProxyExecutor for ATen fallbacks (#112976)

Summary:

Use ProxyExecutor for aten._scaled_dot_product_efficient_attention in ABI-mode

Test Plan:
OSS CI
buck2 run mode/dev-nosan deeplearning/aot_inductor/test:test_custom_ops

Reviewed By: chenyang78, amitaga

Differential Revision: D51005807

jansel commented on torch/_inductor/ir.py:
```diff
@@ -4170,41 +4175,43 @@ def __init__(
-                    else f"at::_ops::{kernel.__name__.replace('.', '_')}::call"
-                )
-                schema = kernel._schema
+                if config.is_fbcode() and kernel in fbcode_use_proxy_executor:
```
Why do we need to diverge behavior between OSS and fbcode here?  It would seem much better to do the same thing in both.

SherlockNoMad replied:
proxy executor doesn't work in OSS yet. 
In fbcode, we use thrift for serialization/deserialization. However, in OSS, we don't have a useable serialization/deserialization solution. 

export uses `json.dumps` for serialization and `json.loads` for deserialization back to python. However, this format is not very friendly for deserialization back to C++.
